frameworks/Rust/actix/src/main.rs

```diff
@@ -0,0 +1,203 @@
+              }
+        )
+        .and_then(move |mut worlds| {
+            // update worlds
+            let mut rng = rand::thread_rng();
+            let between = Range::new(1, 10_000);
+            for world in &mut worlds {
+                world.randomnumber = between.ind_sample(&mut rng);
+            }
+            let body = serde_json::to_string(&worlds).unwrap();
+
+            // persist to db
+            req.state().db.call_fut(db::UpdateWorld(worlds))
+                .from_err()
+                .and_then(move |res| {
+                    if res.is_ok() {
+                        Ok(httpcodes::HTTPOk.build()
+                           .header(header::SERVER, "Actix")
+                           .content_type("application/json")
+                           .content_encoding(headers::ContentEncoding::Identity)
+                           .body(body)?)
+                    } else {
+                        Ok(httpcodes::HTTPInternalServerError.response())
+                    }
+                })
+        })
+        .responder()
+}
+
+#[derive(Template)]
+#[template(path = "fortune.html")]
+struct FortuneTemplate<'a> {
+    items: &'a Vec<models::Fortune>,
+}
+
+fn fortune(req: HttpRequest<State>) -> Box<Future<Item=HttpResponse, Error=Error>> {
+    req.state().db.call_fut(db::TellFortune)
+        .from_err()
+        .and_then(|res| {
+            match res {
+                Ok(rows) => {
+                    let tmpl = FortuneTemplate { items: &rows };
+                    let res = tmpl.render().unwrap();
+
+                    Ok(httpcodes::HTTPOk.build()
+                       .header(header::SERVER, "Actix")
+                       .content_encoding(headers::ContentEncoding::Identity)
+                       .content_type("text/html; charset=utf-8")
+                       .body(res)?)
+                },
+                Err(_) => Ok(httpcodes::HTTPInternalServerError.response())
+            }
+        })
+        .responder()
+}
+
+use std::env;
+fn main() {
+    let sys = System::new("techempower");
+    env::set_var("RUST_BACKTRACE", "1");
+
+    let dbhost = match option_env!("DBHOST") {
+        Some(it) => it,
+        _ => "127.0.0.1"
+    };
+    let db_url = format!(
+        "postgres://benchmarkdbuser:benchmarkdbpass@{}/hello_world", dbhost);
+
+    // Start db executor actors
+    let addr = SyncArbiter::start(
+        num_cpus::get() * 3, move || db::DbExecutor::new(&db_url));
+
+    // start http server
+    HttpServer::new(
+        move || Application::with_state(State{db: addr.clone()})
+            .resource("/json", |r| r.f(json))
+            .resource("/plaintext", |r| r.f(plaintext))
+            .resource("/db", |r| r.route().a(world_row))
+            .resource("/queries", |r| r.route().a(queries))
+            .resource("/fortune", |r| r.route().a(fortune))
+            .resource("/updates", |r| r.route().a(updates)))
+        .backlog(8192)
+        .bind("127.0.0.1:8080").unwrap()
```

Comment:
We employ a host entry for `TFB_SERVER` (or something else) that point to the IP we use in the benchmarks and in the past some implementations have had trouble with `127.0.0.1`; is it possible to bind to something like `0.0.0.0:8080` instead? If not, have you tested with a three machine/vm setup (as is our production benchmark environment) to ensure that the client machine can actually hit this application running on the server?